plugins/woocommerce-blocks/assets/js/blocks/price-filter/block.tsx

```diff
@@ -112,8 +113,14 @@ const PriceFilterBlock = ( {
 	const minPriceParam = getUrlParameter( 'min_price' );
 	const maxPriceParam = getUrlParameter( 'max_price' );
 	const [ queryState ] = useQueryStateByContext();
+	const queryCategory = getSettingWithCoercion(
+		'categoryId',
+		undefined,
+		isNumber
+	);
 	const { results, isLoading } = useCollectionData( {
 		queryPrices: true,
+		...( queryCategory && { queryCategory } ),
 		queryState,
 		isEditor,
 	} );
```

Comment:
`useCollectionData` [supports](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/StoreApi/docs/product-collection-data.md) all product list attributes, including `category`. So instead of modifying the hook, you can pass the category ID directly here, like this:

```js
	const { results, isLoading } = useCollectionData( {
		queryPrices: true,
		queryState: {
			...queryState,
			category: queryCategory,
		},
		isEditor,
	} );

```

Reply:
Tested and working, will push it up